ReactNativeClient/lib/services/rest/Api.js

```diff
@@ -433,7 +433,27 @@ class Api {
 			if (isDataUrl) {
 				await shim.imageFromDataUrl(url, imagePath);
 			} else {
-				await shim.fetchBlob(url, { path: imagePath });
+				const response = await shim.fetchBlob(url, { path: imagePath });
+				let content_type = response.headers['content-type'];
+				if (content_type) {
+					content_type = content_type.split(';')[0]
+				}
+
+				const imageMimeExtensions = {
```

Comment:
Let's re-use Resource.isSupportedImageMimeType for the list of mime types, and mimeUtils.toFileExtension to get the extension.